documentation/docs/06-service-workers.md

```diff
@@ -0,0 +1,13 @@
+---
+title: Service workers
+---
+
+Service workers act as proxy servers that handle network requests inside your app. This makes it possible to make your app work offline, but even if you don't need offline support (or can't realistically implement it because of the type of app you're building), it's often worth using service workers to speed up navigation by precaching your built JS and CSS.
+
+In SvelteKit, if you have a `src/service-worker.js` file (or `src/service-worker.ts`, or `src/service-worker/index.js`, etc) it will be built with Vite and automatically registered.
+
+> You can change the location of your service worker in your [project configuration](#configuration).
+
+Inside the service worker you have access to the [`$service-worker` module](#$service-worker).
+
+Because it needs to be bundled (since browsers don't yet support `import` in this context), and depends on the client-side app's build manifest, **service workers only work in production, not in development**.
```

Comment:
Are there plans to change this? I'm not sure how I'd write code that I can't run until it hits production

Reply:
We probably need a flag to enable them in dev, for local testing.

Reply:
Well presumably you could run a production _build_ locally. (Unless browsers refuse to register them on localhost over HTTP?) This could probably say production build rather than production.

Reply:
Pretty sure localhost is special-cased for service workers and doesn't require https. This should be fine.

Reply:
Updated so that it tells you to use `svelte-kit start` to test the prod build locally. `localhost` is special-cased, though #462 would make that moot anyway